statetransfer/jsondatareader.go

```
@@ -108,44 +87,49 @@ func (m *JsonInitDataReader) Close() error {
 	return nil
 }
 
-type JsonStoredBlockReader struct {
+type JsonRetriableDataReader struct {
 	JsonListReader
 }
 
-func (r *JsonStoredBlockReader) GetNext() (*StoredBlock, error) {
-	if !r.More() {
-		return nil, errNoMore
-	}
-	var elem StoredBlock
-	if err := r.input.Decode(&elem); err != nil {
-		return nil, err
-	}
-	return &elem, nil
+type InitializationDataForRetryableJson struct {
+	Id          common.Hash
+	Timeout     uint64
+	From        common.Address
+	To          common.Address
+	Callvalue   string
+	Beneficiary common.Address
+	Calldata    []byte
 }
 
-func (m *JsonInitDataReader) GetStoredBlockReader() (StoredBlockReader, error) {
-	listreader, err := m.getListReader(m.data.BlocksPath)
-	if err != nil {
-		return nil, err
+func stringToBig(input string) (*big.Int, error) {
+	output, success := new(big.Int).SetString(input, 0)
+	if !success {
+		return nil, fmt.Errorf("%s cannot be parsed as big.Int", input)
 	}
-	return &JsonStoredBlockReader{
-		JsonListReader: listreader,
-	}, nil
-}
-
-type JsonRetriableDataReader struct {
-	JsonListReader
+	return output, nil
 }
 
 func (r *JsonRetriableDataReader) GetNext() (*InitializationDataForRetryable, error) {
 	if !r.More() {
 		return nil, errNoMore
 	}
-	var elem InitializationDataForRetryable
+	var elem InitializationDataForRetryableJson
 	if err := r.input.Decode(&elem); err != nil {
+		panic(fmt.Errorf("decoding retryable: %w", err))
```

Why does this panic instead of returning an error?